python/lsst/pipe/tasks/background.py

```diff
@@ -773,11 +825,310 @@ def getStatsImage(self):
+
+        Returns
+        -------
+        background : `TractBackground`
+            Something guaranteed to be a `TractBackground`.
+        """
+        return cls(other.config, other.tract, other.dims, other.transform, other._values, other._numbers)
+
+    def __init__(self, config, skymap, tract, values=None, numbers=None):
+        """Constructor
+
+        Developers should note that changes to the signature of this method
+        require coordinated changes to the `__reduce__` and `clone` methods.
+
+        Parameters
+        ----------
+        config : `TractBackgroundConfig`
+            Configuration for measuring tract backgrounds.
+        skymap : `lsst.skymap.ringsSkyMap.RingsSkyMap`
+            Skymap object
+        tract : `int`
+            Placeholder Tract ID
+        transform : `lsst.afw.geom.TransformPoint2ToPoint2`
+            Transformation from tract coordinates to warp coordinates.
+        values : `lsst.afw.image.ImageF`
+            Measured background values.
+        numbers : `lsst.afw.image.ImageF`
+            Number of pixels in each background measurement.
+        """
+        self.config = config
+        self.skymap = skymap
+        self.tract = tract
+        self.tractInfo = self.skymap.generateTract(tract)
+        tractDimX, tractDimY = self.tractInfo.getBBox().getDimensions()
+        self.dims = geom.Extent2I(tractDimX / self.config.xBin,
+                                  tractDimY / self.config.yBin)
+
+        if values is None:
+            values = afwImage.ImageF(self.dims)
+            values.set(0.0)
+        else:
+            values = values.clone()
+        assert values.getDimensions() == self.dims
+        self._values = values
+        if numbers is None:
+            numbers = afwImage.ImageF(self.dims)  # float for dynamic range and convenience
+            numbers.set(0.0)
+        else:
+            numbers = numbers.clone()
+        assert numbers.getDimensions() == self.dims
+        self._numbers = numbers
+
+    def __reduce__(self):
+        return self.__class__, (self.config, self.skymap, self.tract, self._values, self._numbers)
+
+    def clone(self):
+        return self.__class__(self.config, self.skymap, self.tract, self._values, self._numbers)
+
+    def addWarp(self, warp):
+        """
+        Equivalent to FocalPlaneBackground.addCcd(), but on warps instead.
+        Bins masked images of warps and adds these values into a blank image
+        with the binned tract dimensions at the location of the warp in the
+        tract.
+
+        Parameters
+        ----------
+        warp : `lsst.afw.image.ExposureF`
+            Warped image corresponding to a single patch in a single visit
+        """
+        image = warp.getMaskedImage()
+        maskVal = image.getMask().getPlaneBitMask(self.config.mask)
+        # Photometric scaling necessary for contiguous background across tract
+        image.image.array *= warp.getPhotoCalib().instFluxToNanojansky(1)
```

Comment:
Prefer `PhotoCalib.calibrateImage(image)`.

That will also scale the variance plane, which this version neglects.

But actually all of our warps now have nJy units already, and I'm not positive that the `PhotoCalib` they have expects positions in tract coordinates, so it might be safest to just assume this has already been done.